scripts/welcome.js

```diff
@@ -13,44 +14,48 @@ var WELCOME_MESSAGES = [
+    return robot.adapter.client.web.conversations.members(room, {limit: API_LIMIT, cursor: cursor})
+        .then(res => getMemberCount(robot, room, memberCount + res.members.length, res.response_metadata.next_cursor));
+}
+
 module.exports = function (robot) {
     robot.enter(function (res) {
         // Make sure we have access to all the clients we need
         if(!robot.adapter.client || !robot.adapter.client.rtm || !robot.adapter.client.web) {
             return;
         }
 
-        // Check the user has entered general
-        var general = robot.adapter.client.rtm.dataStore.getChannelByName("general").id; 
-        if (res.message.room != general) {
+        // Check if user has entered #announcements channel
+        var announcements = robot.adapter.client.rtm.dataStore.getChannelByName("announcements").id; 
+        if (res.message.room != announcements) {
             return;
         }
         
-        // Welcome them to general and send them a personal welcome
-        res.send("Welcome, " + res.message.user.profile.display_name + "!");
+        // Welcome new user to #general
+        var general = robot.adapter.client.rtm.dataStore.getChannelByName("general").id; 
+        name = res.message.user.profile.display_name || res.message.user.name;
+        robot.send({room: general}, "Welcome " + name + "!");
+
+        // Welcome new user personally
         WELCOME_MESSAGES.forEach((message, i) => setTimeout(() => {
             robot.send({room: res.message.user.id}, message);
         }, i * MESSAGE_PAUSE));
 
-        // Check member count to see if we've hit a member milestone
-        robot.adapter.client.web.conversations.members(general).then(result => {
-            // Create a list of promises that resolve to each member's status
-            var memberPromises = result.members.map(id => {
-                return robot.adapter.client.web.users.info(id)
-                    .then(user => (user.deleted) ? 0 : 1);
-                });
-
-            // Get accurate sum of users, filtering out all who are deleted
-            Promise.all(memberPromises)
-                .then(statuses => statuses.reduce((a, b) => a + b, 0))
-                .then(memberCount => {
-                    // If we're not at a member milestone, don't bother celebrating!
-                    if (memberCount % MEMBER_MILESTONE != 0) {
-                        return;
-                    }
-
-                    res.send(":tada: " + memberCount + " members! :tada:");
-                });
-        });
+        // Get member count to see if we've hit a member milestone
+        getMemberCount(robot, announcements, 0).then(memberCount => {
+            // If we're not at a member milestone, don't bother celebrating!
+            if (memberCount % MEMBER_MILESTONE != 0) {
+                return;
+            }
+
+            res.send(":tada: " + memberCount + " members! :tada:");
+        }).catch(err => console.log(err));
```

Comment:
can eta-reduce this to `.catch(console.log)` :P

Reply:
good catch :^) done